Dependencies/AndroidExtensions/Source/JavaWrappers.cpp

```diff
@@ -164,6 +164,36 @@ namespace java::net
     }
 }
 
+namespace android
+{
+    jstring ManifestPermission::getPermissionName(const char* permissionName)
+    {
+        JNIEnv* env{GetEnvForCurrentThread()};
+        jclass cls{env->FindClass("android/Manifest$permission")};
+        jfieldID permId{env->GetStaticFieldID(cls, permissionName, "Ljava/lang/String;")};
+        return (jstring)env->GetStaticObjectField(cls, permId);
+    }
+}
+
+namespace android::app
+{
+    Activity::Activity(jobject object)
+        : Object{"android/app/Activity", object}
+    {
+    }
+
+    void Activity::requestPermissions(const char* permissionName, int permissionRequestID)
+    {
+        jstring systemPermissionName{ManifestPermission::getPermissionName(permissionName)};
+
+        jobjectArray permissionArray{m_env->NewObjectArray(
```

Comment:
Do we need to free this object array after the `CallVoidMethod`?

Reply:
Added call to DeleteLocalRef